pkg/api/api.go

```diff
@@ -17,9 +17,36 @@
 
 package api
 
-const TagYaml = "yaml"
-const TagDoc = "doc"
-const TagEnum = "enum"
+const (
+	FileType              = "file"
```

Comment:
@jotak do you see a way to move all those constantly into the "enum" mechanism  ... doing so will allow us to have automatic documentation of the specific options and make sure we maintain the list going forward.

Context: I was thinking going forward that we will be able to totally deprecate the free-text from the readme and relay on generated docs from the code ( using the `doc` tags in structs).

Reply:
not done yet, it seems to be a bigger refactoring
Currently only the `api` package is covered by doc, right? The idea would be to cover `config` as well? (`config` is like a superset on `api`) Also I'm not sure how the `enum` codegen works, is it run on a specific package or on all sources? Again, it's because these types are in `config` and not in `api`